app/controllers/auth.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+post '/register' do
+  validate_params(params)
+  @user = User.first(email: params[:email])
+  if @user
+    flash[:danger] = 'User already exists. Please Log In.'
+    redirect '/'
+  else
+    @user = User.new email: params[:email]
+    @user.password = @user.hash_password params[:password]
+    @user.save
+
+    if @user.saved?
+      session[:email] = @user.email
+      redirect '/topics'
+    else
+      flash[:danger] = 'Could not create user. Please try again.'
```

Comment:
Since this else part is reused, you can have a method called display_failure

``` ruby
def display_failure(message)
  flash[:danger] = message
  redirect '/
end